deeplearning4j-scaleout/spark/dl4j-spark/src/main/java/org/deeplearning4j/spark/impl/SparkListenable.java

```diff
@@ -0,0 +1,99 @@
+import org.deeplearning4j.api.storage.StatsStorageRouterProvider;
+import org.deeplearning4j.api.storage.listener.RoutingIterationListener;
+import org.deeplearning4j.optimize.api.IterationListener;
+import org.deeplearning4j.spark.api.TrainingMaster;
+import org.deeplearning4j.spark.impl.graph.SparkComputationGraph;
+import org.deeplearning4j.spark.impl.listeners.VanillaStatsStorageRouterProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/** Created by huitseeker on 2/15/17. */
+@Slf4j
+public class SparkListenable implements Serializable {
+
+  protected TrainingMaster trainingMaster;
+  private List<IterationListener> listeners = new ArrayList<>();
+
+  /**
+   * This method allows you to specify IterationListeners for this model.
+   *
+   * @param listeners Iteration listeners
+   */
+  public void setListeners(@NonNull Collection<IterationListener> listeners) {
+    this.listeners.clear();
+    this.listeners.addAll(listeners);
+    if (trainingMaster != null) trainingMaster.setListeners(this.listeners);
+  }
+
+  /**
+   * This method allows you to specify IterationListeners for this model. Note that for listeners
+   * like StatsListener (that have state that will be sent somewhere), consider instead using {@link
+   * #setListeners(StatsStorageRouter, Collection)}
+   *
+   * @param listeners Listeners to set
+   */
+  public void setListeners(@NonNull IterationListener... listeners) {
+    setListeners(Arrays.asList(listeners));
+  }
+
+  /**
+   * Set the listeners, along with a StatsStorageRouter that the results will be shuffled to (in the
+   * case of any listeners that implement the {@link RoutingIterationListener} interface)
+   *
+   * @param statsStorage Stats storage router to place the results into
+   * @param listeners Listeners to set
+   */
+  public void setListeners(StatsStorageRouter statsStorage, IterationListener... listeners) {
+    setListeners(statsStorage, Arrays.asList(listeners));
+  }
+
+  /**
+   * Set the listeners, along with a StatsStorageRouter that the results will be shuffled to (in the
+   * case of any listeners that implement the {@link RoutingIterationListener} interface)
+   *
+   * @param statsStorage Stats storage router to place the results into
+   * @param listeners Listeners to set
+   */
+  public void setListeners(
+      StatsStorageRouter statsStorage, Collection<? extends IterationListener> listeners) {
+    //Check if we have any RoutingIterationListener instances that need a StatsStorage implementation...
+    StatsStorageRouterProvider routerProvider = null;
+    if (listeners != null) {
+      for (IterationListener l : listeners) {
+        if (l instanceof RoutingIterationListener) {
+          RoutingIterationListener rl = (RoutingIterationListener) l;
+          if (rl.getStorageRouter() == null) {
```

Comment:
Condition should be ```statsStorage == null```; whereas ```rl.getStorageRouter()``` being null is totally fine.
(And yes, I realize that's just a refactoring of my code :))